src/cmd/init.rs

```diff
@@ -17,56 +21,69 @@ use super::CmdOutput;
 pub enum NWords {}
 
 #[derive(Clone, Debug, StructOpt)]
-pub struct InitOpt {
-    /// The network name (bitcoin|regtest|testnet)
-    #[structopt(name = "bitcoin|regtest|testnet")]
-    network: Network,
-    /// Existing BIP39 seed words file. Use "-" to read words from stdin.
-    #[structopt(long, name = "FILE")]
-    from_existing: Option<String>,
-    #[structopt(long, default_value = "12", name = "[12|24]")]
-    /// The number of BIP39 seed words to use
-    n_words: usize,
+pub enum InitOpt {
+    /// Initialize a wallet using a seedphrase
+    Seed {
+        /// The network name (bitcoin|regtest|testnet)
+        #[structopt(long, default_value = "bitcoin", name = "bitcoin|regtest|testnet")]
+        network: Network,
+        /// Existing BIP39 seed words file. Use "-" to read words from stdin.
+        #[structopt(long, name = "FILE")]
+        from_existing: Option<String>,
+        #[structopt(long, default_value = "12", name = "[12|24]")]
+        /// The number of BIP39 seed words to use
+        n_words: usize,
+    },
+    /// Initialize using a wallet descriptor
+    Descriptor {
+        /// The network name (bitcoin|regtest|testnet)
+        #[structopt(long, default_value = "bitcoin", name = "bitcoin|regtest|testnet")]
+        network: Network,
+        /// Save unsigned PSBTs to this directory
+        #[structopt(long, parse(from_os_str))]
+        psbt_output_dir: Option<PathBuf>,
+        /// Initialize the wallet from a descriptor
+        #[structopt(name = "wpkh([AAB893A5/84'/0'/0']xpub66..mSXJj/0/*")]
+        external: String,
+        /// Optional change descriptor
+        #[structopt(name = "wpkh([AAB893A5/84'/0'/0']xpub66...mSXJj/1/*)")]
+        internal: Option<String>,
+    },
+    /// Initialize using an xpub
+    Xpub {
```

Comment:
```suggestion
    XPub {
```

Reply:
If I make it XPub then the subcommand looks like `gun init x-pub`, is there a way to stop the kebab case?

Reply:
you can change the name of any variant I think. 
```
#[structopt(name = "xpub")]`
```
If that doesn't work checkout the structopt docs and it shoud tell you